[DOCS][KSPM] KSPM page - new doc (#2330)
elastic/security-docs: docs/cloud-native-security/kspm.asciidoc

[[kspm]]
= Kubernetes security posture management

The Kubernetes Security Posture Management (KSPM) integration allows you to monitor how your Kubernetes clusters' configuration measures up to security benchmarks.

To set up the integration, you'll need to first add it to an agent policy, then deploy the KSPM DaemonSet to the Kubernetes clusters you want to monitor.

[discrete]
== Set up a KSPM integration

> To install the integration:

Reviewer: These steps should be in an ordered list, right? Just checking because they're currently rendering as an unordered list.

> * Go to *Security -> Dashboards -> Cloud Posture*

Reviewer: ```suggestion
. Go to *Security -> Dashboards -> Cloud Posture*.
```

Reviewer: Not sure we need to tell users to navigate to Security. AFAIK, it's assumed they're already within the Security app.

>   * Click *Add a CIS integration*

Reviewer: ```suggestion
  . Click *Add a CIS integration*.
```

3. Click *Add Kubernetes Security Posture Management*.
4. Name your integration.

> 5. Select whether to use the CIS or EKS Benchmarks — use CIS unless you're deploying on EKS.

Author: I'm leaning no, because most people will either know, or all they really need to know is what the second half of the sentence says: use CIS unless you're on EKS.

>   * Select the agent policy where you want to add the integration.

Reviewer: ```suggestion
  * Select the {agent} policy where you want to add the integration.
```

> 7. Click *Save and continue*, then *Add agent to your hosts*.
> 8. The *Add agent* wizard appears, and provides you a DaemonSet manifest `.yaml` with your `Fleet ID`, `Fleet URL`, and other configuration information pre-populated.

Reviewer 2: Step 8 isn't really a separate action, it's just the result of step 7.
```suggestion
7. Click *Save and continue*, then *Add agent to your hosts*. The *Add agent* wizard appears, and provides you a DaemonSet manifest `.yaml` with your `Fleet ID`, `Fleet URL`, and other configuration information pre-populated.
```

> In this case, the agent will be added by deploying a DaemonSet on the Kubernetes clusters you wish to monitor. To do this, for each cluster:

Reviewer: ```suggestion
In this case, the {agent} will be added by deploying a DaemonSet on the Kubernetes clusters you wish to monitor. To do this, for each cluster:
```

Reviewer: Is this linked to the last step in the previous set of steps? If it is, you can add it "under" the first step by using the `+` sign. This will keep it inline with the previous step so it looks like a continuation of the step, and not a new set of instructions.

* Download the manifest and make any necessary revisions to its configuration to suit the needs of your environment.

> * Apply the manifest, for example with `kubectl apply -f elastic-agent-managed-kubernetes.yaml`.

Reviewer: ```suggestion
* Apply the manifest using the `kubectl apply -f ` command, for example: `kubectl apply -f elastic-agent-managed-kubernetes.yaml`
```

> After a minute, the *Add agent* wizard will show “Agent enrollment confirmed”, then “Incoming data confirmed”. You can then click “View assets” to see where the newly-collected configuration information appears throughout {kib}, including the Findings page (see below) and the <<cloud-posture-dashboard, Cloud Posture dashboard>>.

Reviewer: ```suggestion
After a minute, the *Add agent* wizard will show “Agent enrollment confirmed”, then “Incoming data confirmed”. You can then click *View assets* to see where the newly-collected configuration information appears throughout {kib}, including the <<findings-page,Findings page>> and the <<cloud-posture-dashboard, Cloud Posture dashboard>>.
```

> After a minute, the *Add agent* wizard will show “Agent enrollment confirmed”, then “Incoming data confirmed”. You can then click *View assets* to see where the newly-collected configuration information appears throughout {kib}, including the <<findings-page,Findings page>> and the <<cloud-posture-dashboard, Cloud Posture dashboard>>.

Reviewer 2: ```suggestion
After about a minute, the *Add agent* wizard will show “Agent enrollment confirmed”, then “Incoming data confirmed." You can then click *View assets* to see where the newly-collected configuration information appears throughout {kib}, including the <<findings-page,Findings page>> and the <<cloud-posture-dashboard, Cloud Posture dashboard>>.
```

[[findings-page]]
[discrete]

> == The Findings page

Reviewer 2: ```suggestion
== Findings page
```

> image::images/findings-page.png[The Findings page]

Reviewer 2: Add 1 px border to image. Also, is the search bar essential to include in the screenshot? We've cropped similar search bars out of other Security app screenshots, because it's typically a Kibana element that can change without notice.

> The Findings page shows you how the configuration of your Kubernetes clusters matches up to the standards defined on the <<benchmark-rules, CSP Benchmarks page>>.

Reviewer 2: The last part is kind of repetitive with the intro at the top of the page, but if it's going to be that way, use the same phrasing:
```suggestion
The Findings page shows you how the configuration of your Kubernetes clusters measures up to the standards defined on the <<benchmark-rules, CSP Benchmarks page>>.
```

> In the Findings table, findings are organized by the resource IDs of the associated Kubernetes infrastructure, and include data about both the infrastructure and the benchmark rule. Each finding's result (`pass` or `fail`) indicates whether a particular part of your Kubernetes infrastructure meets an active CSP benchmark rule.

Reviewer: ```suggestion
In the Findings table, findings are organized by the resource IDs of the associated Kubernetes infrastructure and include data about the infrastructure and the benchmark rule. Each finding's result (which can be `pass` or `fail`) indicates whether a particular part of your Kubernetes infrastructure meets an active CSP benchmark rule.
```

> In the Findings table, findings are organized by the resource IDs of the associated Kubernetes infrastructure and include data about the infrastructure and the benchmark rule. Each finding's result (which can be `pass` or `fail`) indicates whether a particular part of your Kubernetes infrastructure meets an active CSP benchmark rule.

Reviewer 2: If the page doesn't include anything else besides the Findings table, prob no need to mention it:
```suggestion
Findings are organized by the resource IDs of the associated Kubernetes infrastructure and include data about the infrastructure and benchmark rules. Each finding's result (which can be `pass` or `fail`) indicates whether a particular part of your Kubernetes infrastructure meets an active CSP benchmark rule.
```

> You can filter the data on this page using the search bar at the top.

Reviewer: ```suggestion
You can filter table data by entering queries into the KQL search bar.
```
